consumer/pdata/common.go

```diff
@@ -260,6 +260,14 @@ func (am AttributeMap) InitFromMap(attrMap map[string]AttributeValue) AttributeM
 	return am
 }
 
+// InitEmptyWithCapacity constructs an empty AttributeMap with predefined slice capacity.
+func (am AttributeMap) InitEmptyWithCapacity(cap int) {
```

Comment:
Not sure if this is the right pattern. I was thinking about this for quiet some time, and I was not able to find the best solution. I am trying to achieve something that we allocate the []otlpcommon.AttributeKeyValue as well.

This solves partially the problem, and I am worried that exposing this API now will limit our ability to have the more performant version.